pkg/overcloudipset/configmap.go

```diff
@@ -33,9 +35,20 @@ func CreateConfigMapParams(overcloudIPList ospdirectorv1beta1.OvercloudIPSetList
 	//     https://docs.openstack.org/project-deploy-guide/tripleo-docs/latest/features/deployed_server.html
 	ctlPlaneIps := map[string]string{}
 
+	hostnameMap := make(map[string]string)
+	hostnameFormat := make(map[string]string)
+	roleCount := make(map[string]string)
+
 	for _, ipset := range overcloudIPList.Items {
-		for count := 1; count <= ipset.Spec.HostCount; count++ {
-			hostname := fmt.Sprintf("%s-%d", ipset.Spec.Role, count)
+
+		hostnameFormat[fmt.Sprintf("%sHostnameFormat", ipset.Spec.Role)] = fmt.Sprintf("%s-%%index%%", strings.ToLower(ipset.Spec.Role))
+		roleCount[fmt.Sprintf("%sCount", ipset.Spec.Role)] = strconv.Itoa(ipset.Spec.HostCount)
+
+		for count := 0; count < ipset.Spec.HostCount; count++ {
+			hostname := fmt.Sprintf("%s-%d", strings.ToLower(ipset.Spec.Role), count)
```

Comment:
@dprince could common.CreateOrGetHostname just use the role as input instead of the hostKey using the cr name? then we could use it in all places where we have the roles available.

Reply:
possibly. We may need to change how the data gets stored in the resulting Status block